app/models/order_detail.rb

```diff
@@ -396,6 +397,10 @@ def self.ordered_or_reserved_in_range(start_date, end_date)
     end
   end
 
+  def is_canceled? # non-standard name to avoid conflict with AASM
+    canceled_at.present?
```

Comment:
I really feel like we made a mistake a long time ago making canceled with charge go into "Complete" state, but that's a major refactoring and outside the scope of this work. But something to be aware of.